elasticsearch-transport/lib/elasticsearch/transport/client.rb

```diff
@@ -154,6 +157,11 @@ def initialize(arguments={}, &block)
       #
       def perform_request(method, path, params = {}, body = nil, headers = nil)
         method = @send_get_body_as if 'GET' == method && body
+        if (opaque_id = params.delete(:opaque_id))
+          headers = {} if headers.nil?
+          opaque_id = @opaque_id_prefix ? "#{@opaque_id_prefix}_#{opaque_id}" : opaque_id
```

Comment:
I'm wondering if we shouldn't have the `_` in every prefix join. [Per this comment](https://github.com/elastic/dev/issues/1047#issuecomment-409274544) it looks like the prefix is meant to also contain the separator if one is desired.

Do other clients have prefix functionality?

Reply:
Good catch! Yeah, we shouldn't have the `_` and let the user decide what they want to use :+1:

Reply:
.NET/Go/Python/Php do not support prefix, Javascript does, unsure of perl @elastic/clients-team please confirm :)

@sethmlarson I just noticed the python client allows you to set a global x-opaque-id in the client constructor too. Not sure that quite follows: https://github.com/elastic/clients-team/issues/53

@philkra We need to move that ticket to: 
https://github.com/elastic/clients-team/tree/master/knowledgebase/specifications

Reply:
@Mpdreamz Yeah you're right I misread the .setOpaqueId() section as being a client constructor requirement as well. I need to remove the top-level constructor opaque_id before releasing 7.6.